internal/services/machinelearning/machine_learning_registry_resource.go

```diff
@@ -0,0 +1,449 @@
+				Identity:                      identityIds,
+				Location:                      resp.Model.Location,
+				PublicNetworkAccessEnabled:    pointer.From(prop.PublicNetworkAccess) == "Enabled",
+				Tags:                          pointer.From(resp.Model.Tags),
+				MlFlowRegistryUri:             pointer.From(prop.MlFlowRegistryUri),
+				DiscoveryUrl:                  pointer.From(prop.DiscoveryURL),
+				IntellectualPropertyPublisher: pointer.From(prop.IntellectualPropertyPublisher),
+				ManagedResourceGroup:          pointer.From(pointer.From(prop.ManagedResourceGroup).ResourceId),
+			}
+
+			regions := flattenRegistryRegionDetails(prop.RegionDetails)
+			for i, region := range regions {
+				if i == 0 {
+					model.MainRegion = []ReplicationRegion{region}
+				} else {
+					model.ReplicationRegion = append(model.ReplicationRegion, region)
+				}
+			}
+
+			return metadata.Encode(&model)
+		},
+	}
+}
+
+func (r MachineLearningRegistry) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MachineLearning.RegistryManagement
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model MachineLearningRegistryModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding Machine Learning Registry model %+v", err)
+			}
+
+			id := registrymanagement.NewRegistryID(subscriptionId, model.ResourceGroupName, model.Name)
+			existing, err := client.RegistriesGet(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving existing model for %s", id)
+			}
+
+			param := existing.Model
+
+			if metadata.ResourceData.HasChange("tags") {
+				param.Tags = pointer.To(model.Tags)
+			}
+
+			if metadata.ResourceData.HasChange("identity") {
+				expandedIdentity, err := identity.ExpandLegacySystemAndUserAssignedMapFromModel(model.Identity)
+				if err != nil {
+					return fmt.Errorf("expanding identity: %+v", err)
+				}
+				param.Identity = expandedIdentity
+			}
+
+			if metadata.ResourceData.HasChange("public_network_access_enabled") {
+				param.Properties.PublicNetworkAccess = pointer.To("Disabled")
+				if model.PublicNetworkAccessEnabled {
+					param.Properties.PublicNetworkAccess = pointer.To("Enabled")
+				}
+			}
+
+			if metadata.ResourceData.HasChanges("main_region", "replication_region") {
+				regions := make([]registrymanagement.RegistryRegionArmDetails, 0)
+
+				if len(model.MainRegion) > 0 {
+					regions = append(regions, expandRegistryRegionDetail(model.MainRegion[0]))
+				}
+
+				for _, region := range model.ReplicationRegion {
+					regions = append(regions, expandRegistryRegionDetail(region))
+				}
+
+				param.Properties.RegionDetails = &regions
+			}
```

Comment:
The more I consider this, the less helpful I feel the separation is. 
Can the `Primary` location be anything other than the location (defined at the top level) of the resource? 
If this block is not specified in the create, is there still a block returned?
If only a Primary is specified, does the API accept this, and if so does it provide any value to the user? (i.e. is it in any way meaningful to have "Replication" with no replicas?)